dotCMS/src/main/java/com/dotcms/contenttype/business/StoryBlockAPIImpl.java

```diff
@@ -0,0 +1,268 @@
+package com.dotcms.contenttype.business;
+
+import com.dotcms.content.business.json.ContentletJsonHelper;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.StoryBlockField;
+import com.dotcms.util.ConversionUtils;
+import com.dotmarketing.beans.VersionInfo;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.ContentletVersionInfo;
+import com.dotmarketing.portlets.contentlet.transform.ContentletTransformer;
+import com.dotmarketing.util.Logger;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableList;
+import com.liferay.util.StringPool;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.control.Try;
+import org.apache.commons.lang3.mutable.MutableBoolean;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * default implementation
+ * @author jsanca
+ */
+public class StoryBlockAPIImpl implements StoryBlockAPI {
+
+    @Override
+    public Tuple2<Boolean, Contentlet> refreshReferences(final Contentlet contentlet) {
```

Comment:
I don't know if it is better create a class here to response, something like:

```
class AA {
  boolean refreshed;
  Contentlet contentlet;
...
}
```

I really don't like using Tuple

Reply:
I think I agree on this one. And then when you are getting the values it is difficult to follow the _1(), _2(). I prefer small class with clear getter names.